setup.py

```diff
@@ -29,6 +29,33 @@
 # setuptools
 from setuptools import setup, find_packages
 
+# Specify groups of optional dependencies
+extras_require = {
+    'machine_learning': [
+        'gensim',
+        'numpy',
+        'pyparsing',
+        'python-crfsuite',
+        'scikit-learn',
+        'scipy'
+    ],
+    'plot': [
+        'matplotlib',
+    ],
+    'tests': [
+        'coverage',
+        'nose',
+        'pylint',
+        'tox'
```

Comment:
To run tests you need either tox or coverage+nose+pylint, not both of them. I think removing tox from here is fine.